### ISSUE TITLE
Parsley 2.2.0-rc4 fails with RequireJS

### ISSUE THREAD

**Issue by khalwat** [guillaumepotier/Parsley.js#1022]
I'm sad to report that my initial assessment of Parsley 2.2.0-rc4 working with RequireJS appears to be premature.  Here's a public page where you can see this issue live:

http://imawhale.com/apply

If you click on the "Next" button, most of the time, nothing happens.  The code for the "Next" button is this:

	$('#form-next-button').on('click', function(e) {
	    e.preventDefault();
	      if (false === $('#application-form').parsley().validate('form-section' + (currentSlide + 1) )) 
	        return;
	    $('#form-section' + (currentSlide + 1)).hide();
	    currentSlide++;
	    if (currentSlide > 4)
	    	currentSlide = 4;
	    $('#form-section' + (currentSlide + 1)).fadeIn();
	    updateFormControls();
	});

It dies in the call to `parsley().validate()`

The Parsley script is properly loaded without errors, and everything else works properly with RequireJS... but about 80% of the time when you reload the page, the validate() call above appears to do nothing.

The form validation is initiated by the `data-parsley-validate` on the `<form>` and the field groups are properly tagged with `data-parsley-group` to group them.

All of this works if Parsley is loaded via a `<script>` tag, and not via RequireJS.  Help?

**Comment by khalwat** [comment]
@marcandre For what it's worth, my guess is that because it's sporadic in nature, that the bug it somehow timing related.  Perhaps a race condition somewhere when the script is instantiated via RequireJS?

**Comment by marcandre** [comment]
I took a couple of minutes to look at it.
It's systematic. If you reload when the form is not visible, then scroll to the form, it doesn't work. If you reload when the form is visible, it works.
There's got to be some kind of javascript going on and messing with the form, maybe copying it / detaching it / I don't know for sure, but parsley keeps some references to jQuery elements (`_ui.$errorClassHandler`) and in the cases where it doesn't work, these point to elements that are no longer on the page (same elements, but different `data-parsley-id`).

In short, there's some screwed up interaction with whatever you're using to hide/display the form when scrolling. Maybe binding the form manually at the right time, instead of using `data-parsley-validate`, would resolve your issue. Good luck.

**Comment by khalwat** [comment]
@marcandre Wow, thank you for looking into this.  It was indeed as you expected, apologies for the erroneous report.
